src/core/shape_inference/include/select_shape_inference.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/op/select.hpp>
+
+namespace ov {
+namespace op {
+namespace v1 {
+
+template <class T>
+void shape_infer(const Select* op, const std::vector<T>& input_shapes, std::vector<T>& output_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 3 && output_shapes.size() == 1);
+
+    const auto& broadcast_spec = op->get_auto_broadcast();
+    auto& result_shape = output_shapes[0];
+    if (broadcast_spec.m_type == op::AutoBroadcastType::PDPD) {
+        // PDPD broadcast rule is one-way broadcast. Needs to figure out src and dest
+        // when `then` tensor and `else` tensor are broadcasted to each other.
+        size_t dest_idx = 1;
+        auto try_bidirection = false;
+        auto then_else_merged = false;
+
+        if (input_shapes[1].rank().is_static() && input_shapes[2].rank().is_static()) {
+            auto input1_rank = input_shapes[1].size();
+            auto input2_rank = input_shapes[2].size();
+
+            // When rank is same, need to try broadcast merging after exchange src and dest if default src can't merge
+            // into dest;
+            if (input2_rank == input1_rank)
+                try_bidirection = true;
+            // bigger rank is dest and the other is src;
+            else
+                dest_idx = input1_rank > input2_rank ? 1 : 2;
+        }
+
+        size_t src_idx = (dest_idx == 1) ? 2 : 1;
+        result_shape = input_shapes[dest_idx];
+
+        // Try broadcast merging src into dest
+        then_else_merged = T::broadcast_merge_into(result_shape, input_shapes[src_idx], broadcast_spec);
+
+        // Try the other way if needed
+        if (!then_else_merged && try_bidirection) {
+            result_shape = input_shapes[src_idx];
+            then_else_merged = T::broadcast_merge_into(result_shape, input_shapes[dest_idx], broadcast_spec);
+        }
+
+        NODE_VALIDATION_CHECK(op,
+                              then_else_merged,
+                              "'Else' tensor and `Then` tensor are not broadcasted to each other ");
+        NODE_VALIDATION_CHECK(op,
+                              T::broadcast_merge_into(result_shape, input_shapes[0], broadcast_spec),
+                              "'Cond' tensor shape is not broadcastable.");
```

Comment:
As I can see, broadcasting behavior is changed in comparison with old one for PDPD broadcasting type. The OV documentation also does not define such a way to handle the case when we have PDPD broadcasting.
@luweizhou2016 could you please provide PDPD documentation for Select node where I can find the rules by which the broadcasting is carried out?

Reply:
Please help to check the spec about PDPD broadcast rule in link below. Also in the PartialShape.broadcast_merge_into() implement, for PDPD src shape is expeced to broadcast to des one way in my understanding.
https://github.com/openvinotoolkit/openvino/blob/master/docs/ops/broadcast_rules.md

Reply:
Yes, I agree with your changes in the `PartialShape` and `StaticShape` classes. But I would like to clarify one more point.
In the [current implementation](https://github.com/openvinotoolkit/openvino/blob/master/src/core/src/op/select.cpp#L47) we are doing broadcasting in order `then->else->cond`. In this code you change the order of the broadcast in some cases(because otherwise it will not work). And my question is does the PDPD documentation define this behavior (changing the broadcasting order) for the Select node or not?
If not, shouldn't we keep the current behavior and throw an exception in the case where broadcasting in order `then->else->cond` cannot be done? Thanks!

Reply:
Agree. I have just checked PDPD frontend code. When PDPD frontend construct NGrpaph Select OP, only Numpy broadcast rule is used. So I will leave the code as origininal logic.  Since the  original code logic would have confilcts with the Select OP spec description(then and else broacast to each other), just leave one line comments in the code. Also update some new test cases. Thx!